File: Array + String/leetcode_1806.cpp
#include<bits/stdc++.h>
using namespace std;

bool check(int *array, int n){
    for(int i = 0; i < n; i++){
        if(array[i] != i) return false;
    }
    return true;
}

int reinitializePermutation(int n){
    int *perm = new int[n];
    int *arr = new int[n];
    int operations = 0;

    for(int i = 0; i < n; i++){
        perm[i] = i;
        arr[i] = i;
    }

    for(int i = 0; i < n; i++){
        operations++;
        for(int j = 0; j < n; j++){
            if(j%2 == 0){
                arr[j] = perm[j/2];
            }
            else{
                arr[j] = perm[n / 2 + (j - 1) / 2];
            }
        }
        for(int j = 0; j < n; j++){
            perm[j] = arr[j];
        }
        if(check(perm, n)) return operations;
    }
    return -1;
}

int main(){
    cout<<reinitializePermutation(6)<<endl;
    return 0;
}

File: Array + String/leetcode_1806.js
//check the array with initial one

function check(current){
    for(let i = 0; i < current.length; i++){
        if(current[i] !== i){
            return false
        }
    }
    return true
}


function reinitializePermutation(n){
    let perm = []
    let arr = []
    let operations = 0
    for(let i = 0; i < n; i++){
        perm.push(i)
        arr.push(i)
    }
    for(let i = 0; i < n; i++){
        operations++
        //console.log(perm)
        for(let j = 0; j < n; j++){
            // if(operations === 2){
            //     console.log(perm)
            // }
            if(j % 2 === 0){
                arr[j] = perm[j / 2]
            }
            else{
                arr[j] = perm[n / 2 + (j - 1) / 2]
            }
        }
        perm = arr
        //console.log(perm)
        if(check(perm)){
            return operations
        }
    }
}

console.log(reinitializePermutation(4))